cpp/src/strings/convert/convert_integers.cu

```diff
@@ -38,6 +37,146 @@
+                                       strings.size(),
+                                       cudf::detail::copy_bitmask(strings.parent(), stream, mr),
+                                       strings.null_count(),
+                                       stream,
+                                       mr);
+
+    thrust::transform(rmm::exec_policy(stream),
+                      thrust::make_counting_iterator<size_type>(0),
+                      thrust::make_counting_iterator<size_type>(strings.size()),
+                      results->mutable_view().data<bool>(),
+                      string_to_integer_check_fn<T>{*d_column});
+
+    // Calling mutable_view() on a column invalidates it's null count so we need to set it back
+    results->set_null_count(strings.null_count());
+
+    return results;
+  }
+
+  template <typename T, std::enable_if_t<not std::is_integral<T>::value>* = nullptr>
+  std::unique_ptr<column> operator()(strings_column_view const&,
+                                     rmm::cuda_stream_view,
+                                     rmm::mr::device_memory_resource*) const
+  {
+    CUDF_FAIL("is_integer is expecting an integer type");
+  }
+};
+
+std::unique_ptr<column> is_integer(
+  strings_column_view const& strings,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
+{
+  auto strings_column = column_device_view::create(strings.parent(), stream);
+  auto d_column       = *strings_column;
+  // create output column
+  auto results   = make_numeric_column(data_type{type_id::BOOL8},
+                                     strings.size(),
+                                     cudf::detail::copy_bitmask(strings.parent(), stream, mr),
+                                     strings.null_count(),
+                                     stream,
+                                     mr);
+  auto d_results = results->mutable_view().data<bool>();
+  thrust::transform(rmm::exec_policy(stream),
+                    thrust::make_counting_iterator<size_type>(0),
+                    thrust::make_counting_iterator<size_type>(strings.size()),
+                    d_results,
+                    [d_column] __device__(size_type idx) {
+                      if (d_column.is_null(idx)) return false;
+                      return string::is_integer(d_column.element<string_view>(idx));
+                    });
+
+  // Calling mutable_view() on a column invalidates it's null count so we need to set it back
+  results->set_null_count(strings.null_count());
+
+  return results;
+}
+
+std::unique_ptr<column> is_integer(
+  strings_column_view const& strings,
+  data_type int_type,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
+{
+  if (strings.is_empty()) { return cudf::make_empty_column(data_type{type_id::BOOL8}); }
+  return type_dispatcher(int_type, dispatch_is_integer_fn{}, strings, stream, mr);
+}
+
+}  // namespace
```

Comment:
This needs to move up. The `detail` API should not be inside the anonymous namespace.